packages/create/src/cli/change-data-source/change-data-source.ts

```diff
@@ -0,0 +1,152 @@
+import type { Command } from 'commander';
+import type { AdpProjectData, DataSourceData, PromptDefaults, ChangeDataSourceAnswers } from '@sap-ux/adp-tooling';
+import type { YUIQuestion } from '@sap-ux/inquirer-common';
+import {
+    generateChange,
+    ChangeType,
+    getPromptsForChangeDataSource,
+    getTargetDataSources,
+    getDataSourcesDictionary
+} from '@sap-ux/adp-tooling';
+import { createAbapServiceProvider } from '@sap-ux/system-access';
+import { getLogger, traceChanges } from '../../tracing';
+import prompts from 'prompts';
+import { readFileSync, existsSync } from 'fs';
+import { join } from 'path';
+import { UI5Config } from '@sap-ux/ui5-config';
+import { promptYUIQuestions } from '../../common';
+
+let loginAttempts = 3;
+
+/**
+ * Add a new sub-command to change the data source of an adaptation project to the given command.
+ *
+ * @param {Command} cmd - The command to add the change-data-source sub-command to.
+ */
+export function addChangeDataSourceCommand(cmd: Command): void {
+    cmd.command('change-data-source [path]')
+        .option('-s, --simulate', 'simulate only do not write or install')
+        .action(async (path, options) => {
+            await changeDataSource(path, { ...options }, !!options.simulate);
+        });
+}
+
+/**
+ * Changes the data source of an adaptation project.
+ *
+ * @param {string} basePath - The path to the adaptation project.
+ * @param {PromptDefaults} defaults - The default values for the prompts.
+ * @param {boolean} simulate - if set to true, then no files will be written to the filesystem
+ */
+async function changeDataSource(basePath: string, defaults: PromptDefaults, simulate: boolean): Promise<void> {
+    const logger = getLogger();
+    try {
+        if (!basePath) {
+            basePath = process.cwd();
+        }
+        const configJsonPath = join(basePath, '.adp', 'config.json');
+        if (existsSync(configJsonPath)) {
+            const config = JSON.parse(readFileSync(configJsonPath, 'utf-8'));
+            if (config.environment === 'CF') {
+                throw new Error('Changing data source is not supported for CF projects.');
+            }
+        }
+
+        const variant = JSON.parse(readFileSync(join(basePath, 'webapp', 'manifest.appdescr_variant'), 'utf-8'));
+        const ui5Config = await UI5Config.newInstance(readFileSync(join(basePath, 'ui5.yaml'), 'utf-8'));
+        const { destination, url, client } =
+            ui5Config.findCustomMiddleware<{ backend: Array<{ destination?: string; url?: string; client?: string }> }>(
+                'fiori-tools-proxy'
+            )?.configuration?.backend?.[0] ?? {};
+
+        let target;
+        if (destination) {
+            target = { destination };
+        } else if (url) {
+            target = { url, client };
+        } else {
+            throw new Error('No system configuration found in ui5.yaml');
+        }
+
+        const provider = await createAbapServiceProvider(
```

Comment:
Please use the system-access module here to handle all the different authentication mechanisms and fallbacks.

Reply:
Could you please give a bit more details about your idea here? 
As far as we understand from you comment, we shouldn't do any check whether we have destination or system URL in the ui5.yaml, but directly pass whatever we have to createAbapServiceProvider and let it do the checks for us. Is our understanding correct here?

Reply:
@tobiasqueck Just as a kind reminder, could you please take a look at my previous comment?

Reply:
Sorry, I missed your question. Just checked you code again, you are doing 
```typescript
import { createAbapServiceProvider } from '@sap-ux/system-access';
// ...
const provider = await createAbapServiceProvider(
        target,
        {
            ignoreCertErrors: defaults.ignoreCertErrors
        },
        true,
        logger
    );
```
which is exactly what I want. As result, you will get a valid connection (BAS or VSCode) based on either destinations or stored system configs, or the user is being prompted.

Did you change it already, or did I misread your code when reviewing?

Reply:
As far as I can see the code was there when you did the review. If I understand you correctly, there is not need for a change here?